### PR TITLE
rename TDate to TDateTime

### PR DESCRIPTION
**Pull request by OceanOak**
Changelog:

```
Language 
-  Rename Date to DateTime
```

### CODE REVIEW

**Comment by OceanOak** [comment]
While I was renaming, I came across this comment.
>   // CLEANUP The module is only there for a few functions in the Date module, such as
>   // Date::<. Making these infix wasn't a great idea, and we should remove them.

Should we remove these infix (Date::<, Date::>,etc) functions? If so, what other elements besides the functions themselves , the tests, the regex in the ProgramTypesParser? should we remove?

File: backend/src/LibExecution/RuntimeTypes.fs [hunk 1]
```diff
@@ -331,7 +331,7 @@ and DType =
     | TError -> "Error"
     | THttpResponse _ -> "Response"
     | TDB _ -> "Datastore"
-    | TDate -> "Date"
+    | TDateTime -> "Date"
```
Comment (OceanOak) [inline backend/src/LibExecution/RuntimeTypes.fs:334]: Is this supposed to be changed to  `TDateTime -> "DateTime"`?

Comment (pbiggar) [reply backend/src/LibExecution/RuntimeTypes.fs:334]: yes

Comment (OceanOak) [reply backend/src/LibExecution/RuntimeTypes.fs]: Fixed

**Comment by pbiggar** [comment]
Nice job.

> While I was renaming, I came across this comment.
> 
> > // CLEANUP The module is only there for a few functions in the Date module, such as
> > // Date::<. Making these infix wasn't a great idea, and we should remove them.
> 
> Should we remove these infix (Date::<, Date::>,etc) functions? If so, what other elements besides the functions themselves , the tests, the regex in the ProgramTypesParser? should we remove?

Yeah, this is a good follow-up. All those things sound like the right things to remove.

**Pull request merged**